lms/static/scripts/ui-playground/components/EditAssignment.tsx

```diff
@@ -0,0 +1,348 @@
+import {
+  ArrowLeftIcon,
+  Button,
+  Card,
+  CardActions,
+  CardContent,
+  Checkbox,
+  LinkButton,
+} from '@hypothesis/frontend-shared/lib/next';
+import Library from '@hypothesis/frontend-shared/lib/pattern-library/components/Library';
+import classnames from 'classnames';
+import type { ComponentChildren } from 'preact';
+
+function PickerCard({
+  children,
+  headerContent,
+}: {
+  children: ComponentChildren;
+  headerContent?: ComponentChildren;
+}) {
+  return (
+    <div
+      className={classnames(
+        'bg-grey-1 w-full h-full',
+        // "Real" implementation uses p-2 but this gives more space in demos here
+        'p-2'
+      )}
+    >
+      {headerContent}
+      <div
+        className={classnames(
+          'w-[640px] max-w-[80vw]',
+          'flex flex-col min-h-0 h-full',
+          'mx-auto'
+        )}
+      >
+        <Card classes={classnames('flex flex-col min-h-0')}>{children}</Card>
+      </div>
+    </div>
+  );
+}
+
+function PickerCardHeader({ children }: { children: ComponentChildren }) {
+  return (
+    <div className="bg-slate-0 px-3 py-2 border-b border-slate-5">
+      <h1 className="text-xl text-slate-7 font-normal">{children}</h1>
+    </div>
+  );
+}
+
+function PickerCardContent({ children }: { children: ComponentChildren }) {
+  return (
+    <CardContent size="lg">
+      <div className="grid grid-cols-[auto_1fr] gap-x-6 gap-y-3">
+        {children}
+      </div>
+    </CardContent>
+  );
+}
+
+function PickerCardRow({
+  children,
+  label = '',
+}: {
+  children: ComponentChildren;
+  label?: string;
+}) {
+  return (
+    <>
+      <div className="text-end self-center">
+        <span className="font-medium text-sm leading-none text-slate-7">
+          {label}
+        </span>
+      </div>
+      <div className="space-y-3">{children}</div>
+    </>
+  );
+}
+
+export default function EditAssignmentPage() {
+  return (
+    <Library.Page
+      title="Edit Assignment UI"
+      intro={
+        <>
+          <p>
+            This page documents existing design and UI for configuring
+            assignments and proposes some options for UI enhancement.
+          </p>
+          <p>
+            The assignment-configuration UI was originally designed under the
+            assumption that it would be followed linearly for an assignment that
+            had not yet been configured. With the introduction of support to
+            edit existing assignment configuration, the UI needs to be
+            revisited.
+          </p>
+        </>
+      }
+    >
+      <Library.Section
```

Comment:
Something that might be a little unclear when a new person reads these pages is how the various `Library` components which were originally designed for a hierarchy of `Component group => Component => Usage => Example` map to a hierarchy for workflows / screens. Maybe we could create some aliases or something.